src/Core/Utility/Cache.rei

```diff
@@ -0,0 +1 @@
+let create: (~initialSize: int, 'a => 'b, 'a) => 'b;
```

Comment:
`create` suggests that it's returning a cache object of some kind rather than a function. I think something like `embellish` would be more suggestive of what it actually does.

Or perhaps `Utility.Cache.cached` wouldn't be so bad, if a bit repetitive.

Reply:
`memoize` seems pretty descriptive for this - I switched it to that. Let me know if you prefer `cached` or `embellish` over `memoize` though.

Reply:
Oh yeah, `memoize` is a great name!